staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -143,6 +144,13 @@ func (w *watcher) Watch(ctx context.Context, key string, rev int64, opts storage
 				field.ErrorList{field.Forbidden(field.NewPath("sendInitialEvents"), "for watch is unsupported by the etcd storage because no newListFunc was provided")},
 			)
 		}
+		if w.storage == nil {
```

Comment:
I can move this change to a separate PR if that is easier to review.

Reply:
We could  require a smaller interface. All we need is just one method, see `GetCurrentResourceVersionFromStorage`. It was just easier to pass the whole `w.storage`.